clase 1 19.4/src/main/java/services/FinderClass.java

```diff
@@ -0,0 +1,94 @@
+package services;
+
+import interfaces.IBoardBuilder;
+import interfaces.IFinder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FinderClass implements IFinder {
```

Comment:
You can improve this, using Factory Pattern, like;
`public interface Direction {
   Map<Integer, Integer> getPositions();
}`
`public class NorthDirection implements Direction {
   @Override
   public Map<Integer, Integer> getPositions(){
        Map<Integer, Integer> northPosition = new HashMap<>();
        List<Integer> keys = boardBuilder.putBoardKeysInList();
        int boardSize = (int) Math.round(Math.sqrt(boardBuilder.getBoard().getBoardSize()));

        for (int i = 0; i < keys.size(); i++) {
            for (int j = 0; j < boardSize; j++) {
                if(boardBuilder.getBoard().getBoard().get(i).get(j) == 0){
                    northPosition.put(i-1, j);
                }
            }
        }

       return northPosition;
   }
}`
`public class DirectionFactory {
   public Direction getDirection(String directionString) {
     if (directionString.equals("North")) {
        return new NorthDirection();
     }
   }
}`